### PR TITLE
feat(python): allow for simple creation of n-row empty frame/series via `clear`

### PR DESCRIPTION
**Pull request by alexander-beedie**
Add new param `n` (defaulting to zero, maintaining existing behaviour) to the `.cleared()` method (now renamed/redirected as `.clear()`) for eager/lazy frames and series objects. 

Allows for trivial (and _very_ fast) creation of frames/series of the given size that have been cleared of data, with identical schema to the original. Niche, but handy in some testing scenarios ;)
```python
import polars as pl

df = pl.DataFrame(
    {
        "a": [None, 2, 3, 4],
        "b": [0.5, None, 2.5, 13],
        "c": [True, True, False, None],
    },
    schema_overrides = {"a": pl.UInt32}
)

df.clear()
# ┌─────┬─────┬──────┐
# │ a   ┆ b   ┆ c    │
# │ --- ┆ --- ┆ ---  │
# │ u32 ┆ f64 ┆ bool │
# ╞═════╪═════╪══════╡
# └─────┴─────┴──────┘

df.clear( n=4 )
# ┌──────┬──────┬──────┐
# │ a    ┆ b    ┆ c    │
# │ ---  ┆ ---  ┆ ---  │
# │ u32  ┆ f64  ┆ bool │
# ╞══════╪══════╪══════╡
# │ null ┆ null ┆ null │
# │ null ┆ null ┆ null │
# │ null ┆ null ┆ null │
# │ null ┆ null ┆ null │
# └──────┴──────┴──────┘
```

### CODE REVIEW

**Comment by ritchie46** [comment]
I think it is valuable to add a keyword option `release_data=False`. A slice doesn't release underlying buffers and this might not always be what you want.

**Comment by alexander-beedie** [comment]
> I think it is valuable to add a keyword option `release_data=False`. A slice doesn't release underlying buffers and this might not always be what you want.

Interesting; good catch... this only comes about via the potential `head(0)`? In which case I should probably just avoid doing it that way, rather than add additional complexity/references.

**Comment by alexander-beedie** [comment]
Ok, should be cleaner now (no slice refs).

File: py-polars/polars/internals/dataframe/frame.py [hunk 1]
```diff
@@ -4822,12 +4822,17 @@ def drop_in_place(self, name: str) -> pli.Series:
         """
         return pli.wrap_s(self._df.drop_in_place(name))
 
-    def cleared(self) -> Self:
+    def cleared(self, n: int = 0) -> Self:
```
Comment (ritchie46) [inline py-polars/polars/internals/dataframe/frame.py:4825]: I think we should call this `clear`. We do so with other methods:

`sort -> sorted`
`slice -> sliced`

Comment (alexander-beedie) [reply py-polars/polars/internals/dataframe/frame.py]: Agreed; and done... ;)

**Comment by alexander-beedie** [comment]
This update also helped uncover a small potential edge-case in parametric tests (they should use a string cache context when generating chunked test data, in case of categoricals - fixed).
